enum.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    i = 0
+    new_array = []
+    while i < size
+      new_array << self[i] if yield(self[i]) == true
+      i += 1
+    end
+    new_array
+  end
+
+  def my_all?
```

Comment:
[REQUIRED]

- [ ] my_all? it`s giving an error when a block is passed. Your method should behave like the original one

```
%w[ant bear cat].all? { |word| word.length >= 3 } #=> true
%w[ant bear cat].all? { |word| word.length >= 4 } #=> false
%w[ant bear cat].all?(/t/)                        #=> false
[1, 2i, 3.14].all?(Numeric)                       #=> true
[nil, true, 99].all?                              #=> false
[].all?                                           #=> true
```